pkg/sql/colexec/table_function/meta_scan.go

```diff
@@ -76,7 +76,17 @@ func metaScanCall(_ int, proc *process.Process, arg *Argument) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	rbat = bats[0]
+	metaVecs := make([]*vector.Vector, len(bats[0].Vecs))
+	for i, vec := range bats[0].Vecs {
+		metaVecs[i], err = vec.Dup(proc.Mp())
```

Comment:
Do not duplicate the vector if you will not modify it or you will consume the batch directly.

If the dup is necessary, call the vec.NeedDup() first.